utils/monitoring.js

```diff
@@ -0,0 +1,113 @@
+        labelNames: ['service'],
+        buckets: [1, 10, 60, 600, 3600, 18000, 36000],
+    });
+
+const consolidationDuration = (promClient.register.getSingleMetric('count_items_consolidationDuration'))
+    || new promClient.Histogram({
+        name: 'count_items_consolidationDuration',
+        help: 'Duration of metrics consolidation in seconds',
+        labelNames: ['service'],
+        buckets: [0.01, 0.05, 0.1, 0.2, 0.5, 1, 1.5, 2],
+    });
+
+const workersCount = (promClient.register.getSingleMetric('count_items_workersCountErrors'))
+    || new promClient.Counter({
+        name: 'count_items_workersCountErrors',
+        help: 'Number of errors in counting workers',
+        labelNames: ['state'],
+    });
+
+const objectsCount = (promClient.register.getSingleMetric('count_items_objectsCount'))
+    || new promClient.Counter({
+        name: 'count_items_objectsCount',
+        help: 'Number of objects processed',
+        labelNames: ['state'],
+    });
+
+const metricsCount = (promClient.register.getSingleMetric('count_items_metricsCount'))
+    || new promClient.Counter({
+        name: 'count_items_metricsCount',
+        help: 'Number of entities with data usage',
+        labelNames: ['metricLevel'],
+    });
+/**
+ * @param {http.ServerResponse} res - http response object
+ * @param {Error | errors.ArsenalError} error - Error
+ * @return {void}
+ */
+function _writeResponse(res, error) {
+    let statusCode = 200;
+    if (error) {
+        if (error instanceof errors.ArsenalError && Number.isInteger(error.code)) {
+            statusCode = error.code;
+        } else {
+            statusCode = 500;
+        }
+    }
+    res.writeHead(statusCode, { 'Content-Type': 'application/json' });
+    res.end();
+}
+
+/**
+ * @param {number} countMasterInstance - countMaster instance
+ * @param {function} onScraped - callback to call when metrics are scraped
+ * @param {http.IncomingMessage} req - http request object
+ * @param {http.ServerResponse} res - http response object
+ * @return {void}
+ */
+async function metricsHandler(countMasterInstance, onScraped, req, res) {
+    if (req.method !== 'GET' || req.url !== '/metrics') {
+        return _writeResponse(res, errors.MethodNotAllowed);
+    }
+    try {
+        const [registerMetrics, clusterMetrics] = await Promise.all([
+            promClient.register.metrics(),
+            aggregatorRegistry.clusterMetrics(),
+        ]);
+        const promMetrics = `${registerMetrics}\n${clusterMetrics}`;
+        const contentLen = Buffer.byteLength(promMetrics, 'utf8');
+        return res.writeHead(200, {
+            'content-length': contentLen,
+            'content-type': promClient.register.contentType,
+        }).end(promMetrics);
+    } catch (ex) {
+        return _writeResponse(res, ex);
+    } finally {
+        if (CountMaster.waitingForPromScraping === true) {
+            countMasterInstance.stop(null, onScraped);
+        }
```

Comment:
to reduce coupling (and avoid circular deps between modules), would be best to move this test to the callback:

```suggestion
        if (onScraped) {
            onScraped();
        }
```

Reply:
done here : https://github.com/scality/s3utils/pull/312/commits/be0f3e16b518690540561a94a498ff07dc3860b0#diff-612a4c10e405510cdaad879589d4a7d157df6e0ff42b4645d2a21d4d17e41fa9R74-R76